crates/turborepo-lib/src/run/scope/filter.rs

```diff
@@ -0,0 +1,48 @@
+use tracing::debug;
+use turbopath::{AbsoluteSystemPath, AnchoredSystemPathBuf};
+
+use crate::package_graph;
+
+pub struct PackageInference {
+    package_name: Option<String>,
+    directory_root: AnchoredSystemPathBuf,
+}
+
+impl PackageInference {
+    pub fn calculate(
+        turbo_root: &AbsoluteSystemPath,
+        pkg_inference_path: &AnchoredSystemPathBuf,
+        pkg_graph: &package_graph::PackageGraph,
+    ) -> Self {
+        debug!(
+            "Using {} as a basis for selecting pacakges",
+            pkg_inference_path
+        );
+        let full_inference_path = turbo_root.resolve(pkg_inference_path);
+        for (workspace_name, workspace_entry) in pkg_graph.workspaces() {
+            let pkg_path = turbo_root.resolve(&workspace_entry.package_json_path());
+            let inferred_path_is_below = pkg_path.contains(&full_inference_path);
+            // We skip over the root package as the inferred path will always be below it
+            if inferred_path_is_below && (&pkg_path as &AbsoluteSystemPath) != turbo_root {
+                // set both. The user might have set a parent directory filter,
```

Comment:
This comment always confused me because it makes it sound like this is an error case, but really it's arguably the most common situation, no? Like you're inside a package and you run `turbo` to only run tasks in said package.

Reply:
No, this is the case where you invoke `turbo` in some directory, but _also_ specify a filter with a directory. We'll end up concatenating the two directories. That may or may not select anything, but it's a deterministic calculation.

Reply:
I'm confused, reading the comment, it says "the user might have set a parent directory filter", which made me think this doesn't always happen. So in the case where it doesn't happen, we're just inside a package, right? Like there's no logic in this code that specifically handles the parent directory filter situation?

Reply:
This code is used to calculate the inference values that are used, but is not the code that actually uses them.

The default is the user not setting a directory filter. The only time there is a directory filter is when they pass `--filter={<some/dir>}`. In the case where there is no directory filter set, they may have set some other filter or no filter.

In the case where they have done that, and we calculate an inferred directory, we will append their filter to the end of our inferred directory.